src/api.js

```diff
@@ -1229,6 +1246,130 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
         return this;
     };
 
+    /** Register a custom aggregate with SQLite
+      @example Register a aggregate function
+        db.create_aggregate(
+            "js_sum",
+            function () { return { sum: 0 }; },
+            function (state, value) { state.sum+=value; },
+            function (state) { return state.sum; }
+        );
+        db.exec("CREATE TABLE test (col); INSERT INTO test VALUES (1), (2)");
+        db.exec("SELECT js_sum(col) FROM test"); // = 3
+
+      @param {string} name the name of the aggregate as referenced in
+      SQL statements.
+      @param {object} Aggregate function containing three functions
+      @return {Database} The database object. Useful for method chaining
+       */
+    Database.prototype["create_aggregate"] = function create_aggregate(
+        name,
+        initial_value,
```

Comment:
Shouldn't that have failed in this case already?

```
    result = db.exec("SELECT json_agg(col), json_agg(col2) FROM test2;");
    assert.deepEqual(
        result[0].values[0].map(JSON.parse),
        [["four score", "and seven", "years ago"], [12, 7, 1]],
        "Multiple aggregations at once"
    );
```

both uses of `json_agg` should have been sharing an initial value

Reply:
Yes, my example above actually works because it doesn't mutate the state. But 

```
db.create_aggregate("f", [], { step : (x,y)=> {
    x.push(y);
    return x;
} })
```

would fail

Reply:
ah yeah right I forgot I changed it to allocating a new array instead of pushing. I had fixed it anyway, just was thinking about why that didn't fail